### PR TITLE
implement scan command in redis simulator

### PR DESCRIPTION
**Pull request by foxriver76**
- does the same as keys - this is needed for #464

### CODE REVIEW

File: lib/objects/objectsInMemServerRedis.js [hunk 1]
```diff
@@ -606,34 +606,55 @@ class ObjectsInMemoryServer extends ObjectsInMemoryFileDB {
...
+
+        /**
+         * Get keys matching pattern and send it to given responseId, for "SCAN" and "KEYS" - Objects and files supported
+         *
+         * @param {string} pattern - pattern without namespace prefix
+         * @param {number} responseId - Id where response will be sent to
+         * @param {boolean} isScan - if used by "SCAN" this flag should be true
+         * @private
+         */
+        const _handleScanOrKeys = (pattern, responseId, isScan= false) => {
```
Comment (Apollon77) [inline lib/objects/objectsInMemServerRedis.js:635]: from my point of veiw (also that we do not really have access procection, a normal class method starting with _ would aoso be possible ... when we use  ES6 classes we should avoid "Inline functions" when possible, or?! (discussion topic)

**Comment by foxriver76** [comment]
test will pass with objects-redis 4 its just because test checks err.message but current objects-redis just throws string

File: lib/objects/objectsInMemServerRedis.js [hunk 2]
```diff
@@ -660,13 +681,13 @@ class ObjectsInMemoryServer extends ObjectsInMemoryFileDB {
                             response.push(this.getFileId(entryId, baseName + arr.file, true));
                             response.push(this.getFileId(entryId, baseName + arr.file, false));
                         });
-                        handler.sendArray(responseId, response);
+                        handler.sendArray(responseId, isScan ? ['0', response] : response);
                     });
                 }
             } else {
-                handler.sendError(responseId, new Error('KEYS-UNSUPPORTED for namespace ' + namespace + ': Data=' + JSON.stringify(data)));
+                handler.sendError(responseId, new Error(`KEYS/SCAN-UNSUPPORTED for namespace ${namespace}: Pattern=${pattern}`));
```
Comment (Apollon77) [inline lib/objects/objectsInMemServerRedis.js:688]: Can you please use isScan to output KEY or SCAN at the error message beginning? ;-) It is an implementation detail that we use the same code here, or? ;-)

File: lib/objects/objectsInMemServerRedis.js (hunk 1 above)
Comment (foxriver76) [reply lib/objects/objectsInMemServerRedis.js:635]: I can also make a class method and pass the handler to it if preferred.

**Pull request merged**